pkg/operator/ceph/cluster/mon/mon.go

```diff
@@ -550,9 +554,12 @@ func (c *Cluster) initClusterInfo(cephVersion cephver.CephVersion, clusterName s
 
 	k := keyring.GetSecretStore(c.context, c.ClusterInfo, c.ownerInfo)
 	// store the keyring which all mons share
-	if _, err := k.CreateOrUpdate(keyringStoreName, c.genMonSharedKeyring()); err != nil {
+	secretResourceVersion, err := k.CreateOrUpdate(keyringStoreName, c.genMonSharedKeyring())
```

Comment:
`genMonSharedKeyring()` is only accurate assuming the secret is properly stored into the cluster info secret. It's possible that this isn't the case if a prior reconcile exited after key rotation and before the cluster access secret was updated.

In general, this can be worked around by doing `ceph auth get mon.` to get the true latest mon key. However, this might not work for clusters where mons aren't yet running. @travisn might be able to help clear up all corner cases